pkg/container-utils/cri/cri.go

```diff
@@ -132,7 +182,36 @@ func (c *CRIClient) GetContainerDetails(containerID string) (*runtimeclient.Cont
 		return nil, err
 	}
 
-	return parseContainerDetailsData(c.Name, res.Status, res.Info)
+	podSandbox, err := c.getPodSandboxFromContainerID(containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	return parseContainerDetailsData(c.Name, res.Status, res.Info, podSandbox)
+}
+
+func (c *CRIClient) getPodSandboxFromContainerID(containerID string) (*runtime.PodSandbox, error) {
+	containerID, err := runtimeclient.ParseContainerID(c.Name, containerID)
+	if err != nil {
+		return nil, err
+	}
+
+	containers, err := listContainers(c, &runtime.ContainerFilter{
+		Id: containerID,
+	})
```

Comment:
When called from `WithContainerRuntimeEnrichment()` or from `GetRunningContainers()`, it seems the gRPC method `ListContainers()` will be called for each container. Is it possible to optimize that?

Reply:
I can look into that again.
Reason I have to do this is that `GetContainerDetails` querys the `ContainerStatus`. There we have no information about the `PodSandboxId` that we need.

Reply:
Looking at this again, I see that the gRPC method `ListContainers` will not return all the containers because it is called with a filter. I was afraid that the complexity would be `n^2` with the number of containers. But that's not the case, so I think it is ok.